schema/pgsql/schema.sql

```diff
@@ -56,9 +58,11 @@ CREATE TABLE contact (
     full_name citext NOT NULL,
     username citext, -- reference to web user
     default_channel_id bigint NOT NULL REFERENCES channel(id),
+    external_uuid uuid NOT NULL,
 
-    CONSTRAINT pk_contact PRIMARY KEY (id),
-    UNIQUE (username)
+    UNIQUE (username),
+    UNIQUE (external_uuid),
+    CONSTRAINT pk_contact PRIMARY KEY (id)
```

Comment:
I'm wondering why you reordered the constraints here. Also, in the rest of the schema file, so far the primary key is given first, I'd keep it that way (it also makes sense to me, one can say it's the most important constraint).